_posts/2024-09-26-managing-connector-offsets.md

```diff
@@ -0,0 +1,277 @@
+```
+
+#### Resuming the connector
+
+Once Strimzi has altered the offsets you need to manually resume the connector.
+Edit the `KafkaConnector` resource to set the `state` to `running`:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnector
+# ...
+spec:
+  #...
+  state: running
+  #...
+```
+
+### Resetting offsets
+
+The final action you can perform is to reset the connector offsets.
+For this action the connector also needs to be in a `stopped` state, but you don't need a ConfigMap.
+
+Reset the offsets for your connector by annotating the resource:
+
+```shell
+$ kubectl annotate kafkaconnector my-source-connector strimzi.io/connector-offsets=reset -n kafka
+```
+
+Once completed, the connector offsets will be empty:
+
+```shell
+$ kubectl exec -n kafka -it my-connect-cluster-connect-0 -- curl localhost:8083/connectors/my-source-connector/offsets
+{"offsets":[]}
+```
+
+Similar to altering offsets, make sure you manually resume the connector by changing the `state` in `KafkaConnector` to `running`.
+
+### Managing offsets for MirrorMaker
+
+In addition to connectors managed via the `KafkaConnector` resource, you can also manage the connectors that are deployed as part of a `KafkaMirrorMaker2` resource.
+When using a `KafkaMirrorMaker2` resource, the configurations for the ConfigMaps for listing and altering offsets, as well as the `state` configuration for stopping and resuming the connector, are specified on a per-connector basis. For example:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mirror-maker-2
+spec:
+  # ...
+  mirrors:
+  - # ...
+    sourceConnector:
+      listOffsets:
+        toConfigMap:
+          name: my-connector-offsets
+      alterOffsets:
+        fromConfigMap:
+          name: my-connector-offsets
+      state: stopped
+    # ...
+```
+
+Strimzi allows you to list, alter and reset offsets for all the MirrorMaker connectors (`MirrorSourceConnector`, `MirrorCheckpointConnector`, `MirrorHeartbeatConnector`).
+However, be aware that the only connector that currently actively uses its offsets is the `MirrorSourceConnector`.
+Listing offsets can be useful for the `MirrorCheckpointConnector` and `MirrorHeartbeatConnector` to track progress.
+However, altering or resetting offsets for these connectors is rarely necessary.
+
+Strimzi only allows you to perform actions on one connector within a single mirror at a time.
+Therefore, to initiate an action from a `KafkaMirrorMaker2` resource you must apply two annotations:
+* `strimzi.io/connector-offsets`
+* `strimzi.io/mirrormaker-connector`
+
+Set `strimzi.io/connector-offsets` to one of `list`, `alter` or `reset`.
+At the same time set `strimzi.io/mirrormaker-connector` to the name of your connector.
+
+Strimzi names the connectors using the format `<SOURCE_ALIAS>-><TARGET_ALIAS>.<CONNECTOR_TYPE>`, for example `east-kafka->west-kafka.MirrorSourceConnector`.
```

Comment:
Wondering if something like this:

`[SOURCE_ALIAS]->[TARGET_ALIAS].[CONNECTOR_TYPE]`

is more readable than this:

`<SOURCE_ALIAS>-><TARGET_ALIAS>.<CONNECTOR_TYPE>`

Because of the `->` between `>` and `<`.

Reply:
Good point